cpp/splinepy/splines/helpers/scalar_type_wrapper.hpp

```diff
@@ -31,6 +31,47 @@ void ScalarTypeEvaluate(const SplineType& spline,
   }
 }
 
+/// @brief Evaluate Splines at boundary face centers
+/// output should have size of 2 * para_dim * dim
+template<typename SplineType, typename OutputType>
+void ScalarTypeEvaluateBoundaryCenters(const SplineType& spline,
+                                       OutputType* output) {
+
+  // Prepare inputs
+  const int para_dim = spline.SplinepyParaDim();
+  const int dim = spline.SplinepyDim();
+  const int n_faces = 2 * para_dim;
+
+  double* queries = new double[n_faces * para_dim];
```

Comment:
Please don't do this to me.

```suggestion
  std::vector<double> queries(n_faces * para_dim);
```
If you fear performance issues due to assignments, please use reserve and push_back instead <3

Reply:
I see that you've found the hidden beauty. what about unique_ptr?

Reply:
before everything goes out of hand, let's just go with std::vector